Mutagen.Bethesda.Analyzers.Engine/Reporting/CsvDropbox.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Mutagen.Bethesda.Analyzers.Config;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Plugins.Records;
+namespace Mutagen.Bethesda.Analyzers.Reporting;
+
+public class CsvDropbox : IReportDropbox
+{
+    private readonly IReportDropbox _dropbox;
+    private readonly IReportOutputConfiguration _reportOutputConfiguration;
+
+    public CsvDropbox(
+        IReportDropbox dropbox,
+        IReportOutputConfiguration reportOutputConfiguration)
+    {
+        _dropbox = dropbox;
+        _reportOutputConfiguration = reportOutputConfiguration;
+    }
+
+    public void Dropoff(IModGetter sourceMod, IMajorRecordGetter majorRecord, ITopic topic)
+    {
+        Append(BuildLine(topic, sourceMod, majorRecord));
+        _dropbox.Dropoff(sourceMod, majorRecord, topic);
+    }
+
+    public void Dropoff(ITopic topic)
+    {
+        Append(BuildLine(topic, null, null));
+        _dropbox.Dropoff(topic);
+    }
+
+    private static string BuildLine(ITopic topic, IModGetter? sourceMod, IMajorRecordGetter? majorRecord)
+    {
+        return $"""
+        "{topic.TopicDefinition}","{sourceMod?.ModKey.ToString()}","{majorRecord?.FormKey.ToString()}","{majorRecord?.EditorID}","{topic.FormattedMessage}"
+        """;
+    }
+
+    private void Append(string line)
+    {
+        if (_reportOutputConfiguration.OutputFilePath is null)
```

Comment:
I think this short circuit is too late.   We're doing a lot of work to get to the line string, just to toss it out the window if the feature isn't even active.

I'd go so far as to say we should only register CsvDropbox as a decorator if the feature is active.    In RunAnalyzers, we're building the container and know if the output file is active, and can optionally register at that point